src/lib/ptree/ptree.lua

```diff
@@ -306,6 +340,39 @@ function Manager:start_worker_for_graph(id, graph)
    local actions = self.support.compute_config_actions(
       app_graph.new(), self.workers[id].graph, {}, 'load')
    self:enqueue_config_actions_for_worker(id, actions)
+
+   -- Manage aggregated counters (creation and removal).
+   fiber.spawn(function ()
+      local worker = self.workers[id]
+      local dir = shm.root..'/'..worker.pid
+      local rx = inotify.recursive_directory_inventory_events(dir)
```

Comment:
Here you can create a cond to cancel the stream, and use it later.  Like this:

```lua
local cancel = cond.new()
local rx = inotify.recursive_directory_inventory_events(dir, cancel:wait_operation())
...
```
To cancel the stream, do cond:signal().  it will cause "rm" events to be emitted corresponding to all "creat" events so that all counters will be archived and closed.  Probably you'll want to cancel the events when stopping a worker, and also when shutting down the manager.